### ISSUE TITLE
number parsers

### ISSUE THREAD

**Issue by patch** [patch/cldr-number-pm5#19]
Add number parsers under CLDR::Number::Parse as described in [UTS #35, Part 3, §7](http://www.unicode.org/reports/tr35/tr35-numbers.html#Parsing_Numbers).

**Comment by jochenw** [comment]
If possible, please raise the priority for this one. Parsing is as important as formatting.

**Comment by patch** [comment]
@jochenw: I agree that parsing is important. Most of the features implemented were needed at my work, which are primarily formatting. I don't have time to work on parsing right now, but I'll keep in mind that there's demand for it and will provide updates here with progress. Thank you for your feedback.
